### ISSUE TITLE
Hardcoded number of classes in training loop should be replaced with num_classes

### ISSUE THREAD

**Issue by Mixnikon108** [diffe2023/Diff-E#2]
In the main (#main.py) training loop, there's a line that currently uses a hardcoded number of classes (13) when applying one-hot encoding to the labels:

`y_cat = F.one_hot(y, num_classes=13).type(torch.FloatTensor).to(device)
`

This should be updated to use the num_classes variable instead, to ensure the code works with datasets that have a different number of classes. The corrected line would be:

`y_cat = F.one_hot(y, num_classes=num_classes).type(torch.FloatTensor).to(device)`
